miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/RealMiniApp.kt

```diff
@@ -97,8 +113,16 @@ internal class RealMiniApp(
         }
     }
 
-    override suspend fun getMiniAppManifest(appId: String, versionId: String): MiniAppManifest =
-        miniAppDownloader.fetchMiniAppManifest(appId, versionId)
+    override suspend fun getMiniAppManifest(appId: String, versionId: String): MiniAppManifest {
+        val latestManifest = miniAppDownloader.fetchMiniAppManifest(appId, versionId)
+        // store the latest manifest value in cache after fetched from api
+        miniAppManifestCache.storeApiManifest(appId, latestManifest)
```

Comment:
This is only storing the App ID and not the version, so we still don't have a guarantee that the manifest is for the correct version. Could you also cache the version?

Also, I'm not sure if `MiniAppManifestCache` is the right place to cache this. It's unrelated to the functionality of our manifest cache - it's only about saving us from hitting the API multiple times in a row for the same thing. So maybe it should be moved into `MiniAppDownloader`? And then we can cache only one manifest: the last one that we retrieved from the API (along with the app id and version id).

That also saves us from having to add extra calls to `miniAppManifestCache` - we can just call `miniAppDownloader.fetchMiniAppManifest`, and if it has the cached version then it returns the cached manifest, otherwise it downloads it.

What do you think?

Reply:
Yes, I used version id too to store the manifest from api, so its saving us from hitting the api unnecessarily.